storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/StorIOSQLiteProcessor.java

```diff
@@ -95,41 +96,46 @@ protected void processAnnotatedFieldsOrMethods(@NotNull final RoundEnvironment r
-            }
+                if (storIOSQLiteTypeMeta == null) {
+                    throw new ProcessingException(annotatedFieldElement, "Field marked with "
+                            + StorIOSQLiteColumn.class.getSimpleName()
+                            + " annotation should be placed in class marked by "
+                            + StorIOSQLiteType.class.getSimpleName()
+                            + " annotation"
+                    );
+                }
 
-            // If field annotation applied to both fields and methods in a same class.
-            if ((storIOSQLiteTypeMeta.needCreator && !storIOSQLiteColumnMeta.isMethod()) ||
-                    (!storIOSQLiteTypeMeta.needCreator && storIOSQLiteColumnMeta.isMethod() && !storIOSQLiteTypeMeta.columns.isEmpty())) {
-                throw new ProcessingException(annotatedFieldElement, "Can't apply"
-                        + StorIOSQLiteColumn.class.getSimpleName()
-                        + " annotation to both fields and methods in a same class"
-                );
-            }
+                // If class already contains column with same name -> throw an exception.
+                if (storIOSQLiteTypeMeta.columns.containsKey(storIOSQLiteColumnMeta.storIOColumn.name())) {
+                    throw new ProcessingException(annotatedFieldElement, "Column name already used in this class");
+                }
 
-            // If column needs creator then enclosing class needs it as well.
-            if (!storIOSQLiteTypeMeta.needCreator && storIOSQLiteColumnMeta.isMethod()) {
-                storIOSQLiteTypeMeta.needCreator = true;
-            }
+                // If field annotation applied to both fields and methods in a same class.
+                if ((storIOSQLiteTypeMeta.needCreator && !storIOSQLiteColumnMeta.isMethod()) ||
+                        (!storIOSQLiteTypeMeta.needCreator && storIOSQLiteColumnMeta.isMethod() && !storIOSQLiteTypeMeta.columns.isEmpty())) {
+                    throw new ProcessingException(annotatedFieldElement, "Can't apply"
+                            + StorIOSQLiteColumn.class.getSimpleName()
+                            + " annotation to both fields and methods in a same class"
+                    );
+                }
+
+                // If column needs creator then enclosing class needs it as well.
+                if (!storIOSQLiteTypeMeta.needCreator && storIOSQLiteColumnMeta.isMethod()) {
+                    storIOSQLiteTypeMeta.needCreator = true;
+                }
 
-            // Put meta column info.
-            storIOSQLiteTypeMeta.columns.put(storIOSQLiteColumnMeta.storIOColumn.name(), storIOSQLiteColumnMeta);
+                // Put meta column info.
+                storIOSQLiteTypeMeta.columns.put(storIOSQLiteColumnMeta.storIOColumn.name(), storIOSQLiteColumnMeta);
+            } catch (SkipNotAnnotatedClassWithAnnotatedParentException e) {
+                e.printStackTrace();
```

Comment:
I guess that would be better :) What about logging it via annotation processor logger (forgot its name) with warning/etc level?